src/protocol.rs

```diff
@@ -101,6 +101,30 @@ pub enum Error {
     Internal,
 }
 
+impl From<HpkeError> for Error {
+    fn from(e: HpkeError) -> Self {
+        Self::Hpke(HpkeErrorWrapper(e))
+    }
+}
+
+/// This is a wrapper for HpkeError, as the type doesn't support Eq.
```

Comment:
Maybe try open an upstream PR? Adding PartialEq or kind() method seems reasonable.

Reply:
Yeah, it's in the plan: https://github.com/rozbb/rust-hpke/pull/13